MAINT: Bump to 1.10.2 and simplify noarch approach

larsoner opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

Closes #146

larsoner committed:
MAINT: Bump to 1.10.2

larsoner committed:
FX: Ver

larsoner commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18481611497](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18481611497). Examine the logs at this URL for more detail.</sub>

Commit:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.13.17.18.1

larsoner committed:
WIP: Better

larsoner committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.14.10.19.06

Other tools:
- conda-build 25.5.0
- rattler-build 0.33.0
- rattler-build-conda-compat 1.4.9

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`, or selector on line 93:
        - pyqt  # [build_variant == "pyqt"]


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18497548272](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18497548272). Examine the logs at this URL for more detail.</sub>

larsoner committed:
FIX: Name

larsoner committed:
Revert "FIX: Name"

This reverts commit bed6ce3af15110a2d9c8a556f9b4c10effce4ab9.

larsoner committed:
Revert "MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.14.10.19.06"

This reverts commit 1ec26037af10fdcd17874989c0b98dde5432680a.

larsoner committed:
Revert "WIP: Better"

This reverts commit d56534beae243f140b2dd5588774614dfe3068af.

larsoner committed:
FIX: Maybe

larsoner committed:
WIP

larsoner committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.14.10.19.06

Other tools:
- conda-build 25.5.0
- rattler-build 0.33.0
- rattler-build-conda-compat 1.4.9

larsoner committed:
FIX: Skip

larsoner committed:
FIX: Mabye

larsoner committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.14.10.19.06

Other tools:
- conda-build 25.5.0
- rattler-build 0.33.0
- rattler-build-conda-compat 1.4.9

larsoner committed:
FIX: YAML true

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18498929899](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18498929899). Examine the logs at this URL for more detail.</sub>

larsoner committed:
FIX: Restore

larsoner committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.14.10.19.06

Other tools:
- conda-build 25.5.0
- rattler-build 0.33.0
- rattler-build-conda-compat 1.4.9

larsoner committed:
FIX: Maybe

larsoner commented:
Okay artifacts list from looking at the builds:
```
mne-1.10.2-linux_pyqt_h499dab4_200.conda
mne-base-1.10.2-pyha770c72_200.conda
mne-1.10.2-linux_pyside6_h60f6096_100.conda
mne-base-1.10.2-pyha770c72_100.conda
mne-1.10.2-osx_pyqt_h8d421b4_200.conda
mne-base-1.10.2-pyh694c41f_200.conda
mne-1.10.2-osx_pyside6_h94d15eb_100.conda
mne-base-1.10.2-pyh694c41f_100.conda
```
which matches with https://anaconda.org/conda-forge/mne/files so I think we're good to go!

hoechenberger commented on recipe/meta.yaml:
```diff
@@ -21,6 +21,7 @@ source:
 
 build:
   number: {{ build }}
+  skip: true  # [py>310]  # only need to build on 3.10
```
interesting, since when is this necessary? I thought in the past a noarch build always only took place with the oldest Python version.

larsoner replied:
For as long as I can remember, the build matrix had all python versions, e.g., 3.10, 3.11, 3.12, 3.13. But they were all redundant. You can probably look through the CI logs to figure out when it started doing the whole matrix.

larsoner replied:
... and I tried a ton of things to get it down to just one Python version -- this is what finally worked.

Putting `noarch: python` at the top-level `build:` did cause a single run / python version to spawn. But then it complained about our version selectors in the linter, so that approach wouldn't work...

hoechenberger replied:
ahh!  right 

now I understand

thanks for the clarification, Eric!
